shared-bindings/tca9555/__init__.h

```diff
@@ -0,0 +1,119 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef MICROPY_INCLUDED_SHARED_BINDINGS_TCA9555___INIT___H
+#define MICROPY_INCLUDED_SHARED_BINDINGS_TCA9555___INIT___H
+
+#include "common-hal/microcontroller/Pin.h"
+
+// Type object used in Python. Should be shared between ports.
+extern const mp_obj_dict_t tca_pin_globals;
+extern const mp_obj_type_t tca_pin_type;
+
+const mcu_pin_obj_t *validate_obj_is_free_pin_including_tca(mp_obj_t obj, qstr arg_name);
+const mcu_pin_obj_t *validate_obj_is_free_pin_including_tca_or_none(mp_obj_t obj, qstr arg_name);
+const mcu_pin_obj_t *validate_obj_is_pin_including_tca(mp_obj_t obj, qstr arg_name);
+
+extern bool common_hal_tca_gpio_get_input(uint tca_gpio);
+extern bool common_hal_tca_gpio_get_output(uint tca_gpio);
+extern bool common_hal_tca_gpio_get_config(uint tca_gpio);
+extern bool common_hal_tca_gpio_get_polarity(uint tca_gpio);
+
+extern void common_hal_tca_gpio_set_output(uint tca_gpio, bool value);
+extern void common_hal_tca_gpio_set_config(uint tca_gpio, bool output);
+extern void common_hal_tca_gpio_set_polarity(uint tca_gpio, bool polarity);
+
+extern uint16_t common_hal_tca_get_input_port(uint tca_index);
+extern uint8_t common_hal_tca_get_input_port_low(uint tca_index);
+extern uint8_t common_hal_tca_get_input_port_high(uint tca_index);
+
+extern uint16_t common_hal_tca_get_output_port(uint tca_index);
+extern uint8_t common_hal_tca_get_output_port_low(uint tca_index);
+extern uint8_t common_hal_tca_get_output_port_high(uint tca_index);
+
+extern uint16_t common_hal_tca_get_config_port(uint tca_index);
+extern uint8_t common_hal_tca_get_config_port_low(uint tca_index);
+extern uint8_t common_hal_tca_get_config_port_high(uint tca_index);
+
+extern uint16_t common_hal_tca_get_polarity_port(uint tca_index);
+extern uint8_t common_hal_tca_get_polarity_port_low(uint tca_index);
+extern uint8_t common_hal_tca_get_polarity_port_high(uint tca_index);
+
+extern void common_hal_tca_set_output_port(uint tca_index, uint16_t output_state);
+extern void common_hal_tca_set_output_port_low(uint tca_index, uint8_t output_state);
+extern void common_hal_tca_set_output_port_high(uint tca_index, uint8_t output_state);
+
+extern void common_hal_tca_set_config_port(uint tca_index, uint16_t config_state);
+extern void common_hal_tca_set_config_port_low(uint tca_index, uint8_t config_state);
+extern void common_hal_tca_set_config_port_high(uint tca_index, uint8_t config_state);
+
+extern void common_hal_tca_set_polarity_port(uint tca_index, uint16_t polarity_state);
+extern void common_hal_tca_set_polarity_port_low(uint tca_index, uint8_t polarity_state);
+extern void common_hal_tca_set_polarity_port_high(uint tca_index, uint8_t polarity_state);
```

Comment:
All of these should be on a TCA9555 object because `tca_index` (I think) is which chip you are talking to. The TCA9555 object should take in the I2C bus it is on and also the device address.